src/NuGet.Core/NuGet.Packaging/Signing/Archive/CentralDirectoryHeader.cs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+
+namespace NuGet.Packaging.Signing
+{
+    internal sealed class CentralDirectoryHeader
+    {
+        internal const uint Signature = 0x02014b50;
+
+        internal ushort VersionMadeBy { get; private set; }
+        internal ushort VersionNeededToExtract { get; private set; }
+        internal ushort GeneralPurposeBitFlag { get; private set; }
+        internal ushort CompressionMethod { get; private set; }
+        internal ushort LastModFileTime { get; private set; }
+        internal ushort LastModFileDate { get; private set; }
+        internal uint Crc32 { get; private set; }
+        internal uint CompressedSize { get; private set; }
+        internal uint UncompressedSize { get; private set; }
+        internal ushort FileNameLength { get; private set; }
+        internal ushort ExtraFieldLength { get; private set; }
+        internal ushort FileCommentLength { get; private set; }
+        internal ushort DiskNumberStart { get; private set; }
+        internal ushort InternalFileAttributes { get; private set; }
+        internal uint ExternalFileAttributes { get; private set; }
+        internal uint RelativeOffsetOfLocalHeader { get; private set; }
+        internal byte[] FileName { get; private set; }
+        internal byte[] ExtraField { get; private set; }
+        internal byte[] FileComment { get; private set; }
+
+        internal uint GetSize()
```

Comment:
given that this size doesn't include signature, should we change the name or add a comment to specify that is size of the header without signature?

Reply:
Actually, `SignedPackageArchiveIOUtility.CentralDirectoryFileHeaderSizeWithoutSignature` is misnamed.  Its value (46) includes the signature.  I'll fix this though.

```
Field                             Bytes
--------------------------------- ------
uint Signature                    4
ushort VersionMadeBy              2
ushort VersionNeededToExtract     2
ushort GeneralPurposeBitFlag      2
ushort CompressionMethod          2
ushort LastModFileTime            2
ushort LastModFileDate            2
uint Crc32                        4
uint CompressedSize               4
uint UncompressedSize             4
ushort FileNameLength             2
ushort ExtraFieldLength           2
ushort FileCommentLength          2
ushort DiskNumberStart            2
ushort InternalFileAttributes     2
uint ExternalFileAttributes       4
uint RelativeOffsetOfLocalHeader  4
--------------------------------- ------
Total                             46
```